module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php

```diff
@@ -0,0 +1,260 @@
+<?php
+/**
+ * Record driver data formatting view helper
+ *
+ * PHP version 5
+ *
+ * Copyright (C) Villanova University 2016.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+use VuFind\RecordDriver\AbstractBase as RecordDriver;
+use Zend\View\Helper\AbstractHelper;
+
+/**
+ * Record driver data formatting view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class RecordDataFormatter extends AbstractHelper
+{
+    /**
+     * Default settings.
+     *
+     * @var array
+     */
+    protected $defaults = [];
+
+    /**
+     * Create formatted key/value data based on a record driver and field spec.
+     *
+     * @param RecordDriver $driver Record driver object.
+     * @param array        $spec   Formatting specification
+     *
+     * @return Record
+     */
+    public function getData(RecordDriver $driver, array $spec)
+    {
+        $result = [];
+
+        // Sort the spec into order by position:
+        $sort = function ($a, $b) {
+            $posA = isset($a['pos']) ? $a['pos'] : 0;
+            $posB = isset($b['pos']) ? $b['pos'] : 0;
+            if ($posA === $posB) {
+                return 0;
+            }
+            return $posA < $posB ? -1 : 1;
+        };
+        uasort($spec, $sort);
+
```

Comment:
Might be overkill but maybe the sort function should go into its own method, just in case someone wants to override it?